gslib/commands/iam.py

```diff
@@ -528,7 +636,14 @@ def _PatchIamHelperInternal(self,
     orig_bindings = list(bindings)
 
     for (is_grant, diff) in bindings_tuples:
-      bindings = PatchBindings(bindings, BindingsTuple(is_grant, diff))
```

Comment:
Did you change the inputs?

Reply:
I did. You'll see this when you get to iam_helper, but previously PatchBindings took a bindings message, passed it to BindingsToDict which transformed it to a `role: set(members)` dict which is convenient for updates. Then, PatchBindings would perform the updates, and convert the bindings back to a bindings message.

The shim doesn't have much use for the bindings messages, since we pass and receive strings directly from subcommands, so changing PatchBindings to use `role: set(members)` dicts directly cut down on a lot of unnecessary conversions.

Reply:
Weren't we planning on deprecating Apitools completely? We might not have time to do that, but would it be simpler to make dict-only tools instead of mixing this logic?

Reply:
I guess I didn't quite complete the thought in the earlier comment. The changes to PatchBindings do move it towards only using dicts. It no longer takes apitools messages.